lib/active_merchant/billing/gateways/place_to_pay.rb

```diff
@@ -0,0 +1,235 @@
+        else
+          return response['status']['status'];
+        end
+      end
+
+      def message_from(response)
+        response['status'] && response['status']['message'] ? response['status']['message'] :
+          COMMON_MESSAGES[:missed]
+      end
+
+      def authorization_from(response); end
+
+      def post_data(action, parameters = {})
+        JSON.generate(parameters)
+      end
+
+      def error_code_from(action, response, options)
+        unless success_from(action, response, options)
+          # TODO: lookup error code for this response
+          STANDARD_ERROR_CODE_MAPPING[response['reason']]
+        end
+      end
+
+      def add_auth(post, options)
+        original_nonce = generate_nonce()
+        seed = get_date_in_iso_format()
+        tranKey = generate_trans_key(original_nonce, seed)
+        post[:auth] = {
+          login: @login,
+          tranKey: tranKey,
+          seed: seed,
+          nonce: convert_to_Base64(original_nonce)
+        }
+      end
+
+      def add_payer(post, options)
+        post[:payer] = options[:payer]
+      end
+
+      def add_payment(post, options)
+        post[:payment] = options[:payment]     
+      end
+
+      def add_instrument(post, payment, options)
+        post[:instrument] = options[:instrument]
+        post[:instrument][:card][:number] = payment.number;
+        post[:instrument][:card][:expiration] = (payment.month).to_s + "/" + (payment.month).to_s;
+        post[:instrument][:card][:cvv] = payment.verification_value;        
+      end
+
+      def add_amount(post, payment, options)
+        post[:amount] = {
+          currency: options[:currency], #Allowed values: USD COP CRC EUR CAD AUD GBP MXN CLP
+          total: options[:total]
+        }
+      end
+
+      def generate_nonce
+        Digest::MD5.hexdigest(rand(10).to_s)
+      end
+
+      def get_date_in_iso_format
+        Time.now.strftime(ISO_FORMAT_DATE)
+      end      
+
+      def convert_to_Base64(value)
+        Base64.strict_encode64(value).chomp
+      end
+
+      def generate_trans_key(original_nonce, seed)
+        base_tran_key = "#{original_nonce}#{seed}#{@secret_key}"
+        key_to_digest = Digest::SHA256.digest(base_tran_key)
+        convert_to_Base64(key_to_digest)
+      end
+
+      def request_headers()
+        headers = {'Content-Type' => 'application/json'}
+        headers
+      end
+
+      def api_request(method, url, rel_path, params)
+        params == {} ? body = '' : body = params.to_json
+
+        raw_response = ssl_request(method, url, body, request_headers())
+        # puts "Raw response: "
+        # puts JSON.pretty_generate(raw_response)
+
+        response = parse(raw_response)
+        
+        # puts "Response: "
+        # puts JSON.pretty_generate(response)
+      
+        return response
+      end
+
+      def parse(body)
+        return {} if body.empty? || body.nil?
```

Comment:
It looks like you have two definitions of `parse`, the first one should be removed.